go/pools/resource_pool.go

```diff
@@ -333,6 +333,9 @@ func (rp *ResourcePool) getWithSettings(ctx context.Context, settings []string)
 
 	if !wrapper.resource.IsSettingsApplied() {
 		if err = wrapper.resource.ApplySettings(ctx, settings); err != nil {
+			// as we are not able to apply settings, we can return this connection to non-settings channel.
+			// TODO: may check the error code to see if it is recoverable or not.
+			rp.resources <- wrapper
```

Comment:
Is there anything to reset on the wrapper so it can be cleanly reused?

Reply:
There is nothing here to clean as of now. It comes here only when no settings are applied earlier and now the current settings failed so it can be reused.

Reply:
Got it